client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -20,13 +21,42 @@ private final class PoolManager[F[_], A <: Connection[F]](
   private[this] val logger = getLogger(classOf[PoolManager[F, A]])
 
   private var isClosed = false
-  private var allocated = 0
-  private val idleQueue = new mutable.Queue[A]
-  private val waitQueue = new mutable.Queue[Waiting]
+  private var curTotal = 0
+  private var allocated = mutable.Map.empty[RequestKey, Int]
+  private val idleQueue = mutable.Map.empty[RequestKey, mutable.Queue[A]]
+  private val waitQueue = mutable.Queue.empty[Waiting]
 
   private def stats =
     s"allocated=$allocated idleQueue.size=${idleQueue.size} waitQueue.size=${waitQueue.size}"
 
+  @inline
```

Comment:
I almost never use this annotation.  The JVM automates it better than I can guess it.  Do you have evidence this is helpful?

Reply:
Just a request for scala compiler to consider inlining, we can leave it on, no?